Plutarch/Builtin/ByteString.hs

```diff
@@ -47,6 +83,33 @@ instance Semigroup (Term s PByteString) where
 instance Monoid (Term s PByteString) where
   mempty = punsafeConstantInternal $ PLC.someValue BS.empty
 
+{- | Type designating whether a conversion should be most-significant-first or
+most-significant-last. See
+[CIP-121](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0121#representation)
+for more details on this.
+
+@since WIP
+-}
+newtype PEndianness (s :: S) = PEndianness (Term s PBool)
+  deriving stock
+    ( -- | @since WIP
+      Generic
+    )
+
+{- | Indicates the conversion should be most-significant-first.
+
+@since WIP
+-}
+pmostSignificantFirst :: forall (s :: S). Term s PEndianness
+pmostSignificantFirst = punsafeCoerce ptrue
+
+{- | Indicates the conversion should be most-significant-last.
+
+@since WIP
+-}
+pmostSignificantLast :: forall (s :: S). Term s PEndianness
+pmostSignificantLast = punsafeCoerce pfalse
```

Comment:
```suggestion
pmostSignificantFirst :: forall (s :: S). Term s PEndianness
pmostSignificantFirst = pcon $ PEndianness ptrue

{- | Indicates the conversion should be most-significant-last.

@since WIP
-}
pmostSignificantLast :: forall (s :: S). Term s PEndianness
pmostSignificantLast = pcon $ PEndianness pfalse
```
No need for unsafe

Reply:
Dependency cycle.